src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -385,6 +385,21 @@ def get_enable_artifact_streaming(self) -> bool:
                 enable_artifact_streaming = self.agentpool.artifact_streaming_profile.enabled
         return enable_artifact_streaming
 
+    def get_pod_ip_allocation_mode(self: bool = False) -> Union[str, None]:
+        """Get the value of pod_ip_allocation_mode.
+        :return: str or None
+        """
+        # Initialize the value of pod_ip_allocation_mode from the
+        # agentpool.network_profile.pod_ip_allocation_mode if it exists
+        pod_ip_allocation_mode = None
+        if (self.agentpool and self.agentpool.pod_ip_allocation_mode is not None):
+            pod_ip_allocation_mode = self.agentpool.pod_ip_allocation_mode
+
+        # overwrite if provided by user in the command
+        pod_ip_allocation_mode = self.raw_param.get("pod_ip_allocation_mode")
```

Comment:
This will set the value to user input (if use does not specify the option, default value is `None`) no matter what the value is stored in managed cluster/agent pool object, in other words line 394-396 is meanlingless.

Reply:
Good catch. Will change the order of this logic. Updated the method to
```
def get_pod_ip_allocation_mode(self: bool = False) -> Union[str, None]:
        """Get the value of pod_ip_allocation_mode.
        :return: str or None
        """

        # Get the value of pod_ip_allocation_mode from the raw parameters provided by the user
        pod_ip_allocation_mode = self.raw_param.get("pod_ip_allocation_mode")
        # In create mode, try to read the property value corresponding to the parameter from the `agentpool` object
        # if it exists and user has not provided any value in raw parameters
        if self.decorator_mode == DecoratorMode.CREATE:
            if (
                pod_ip_allocation_mode and 
                self.agentpool and 
                self.agentpool.pod_ip_allocation_mode is not None
            ):
                pod_ip_allocation_mode = self.agentpool.pod_ip_allocation_mode

        return pod_ip_allocation_mode
        ```